server/src-lib/Hasura/RQL/DDL/Remote/Validate.hs

```diff
@@ -141,38 +144,50 @@ validateRemoteArguments expectedArguments providedArguments permittedVariables =
 
 -- | Validate a value against a type.
 validateType ::
-     HashMap G.Variable FieldInfo
+     HM.HashMap G.Variable FieldInfo
   -> G.Value
   -> G.GType
+  -> HM.HashMap G.NamedType TypeInfo
   -> Validation (NonEmpty ValidationError) ()
-validateType permittedVariables value (gtypeToEither -> expectedType) =
```

Comment:
This makes `gtypeToEither` now dead code, so you can remove `gtypeToEither`.

Reply:
Actually, we will need it because currently I am not distinguishing list types from non-list types.

So for an argument like
```
message(id: $id) {
  leaf1
}
```
This will validate for remote argument of the form `id: [$id]` also. Will do this check in the next commit.